nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2018-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.helm.internal.content.createindex;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.common.event.EventAware.Asynchronous;
+import org.sonatype.nexus.common.event.EventManager;
+import org.sonatype.nexus.common.stateguard.Guarded;
+import org.sonatype.nexus.repository.FacetSupport;
+import org.sonatype.nexus.repository.Repository;
+import org.sonatype.nexus.repository.manager.RepositoryCreatedEvent;
+import org.sonatype.nexus.repository.view.Content;
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.repository.helm.internal.content.HelmContentFacet;
+import org.sonatype.repository.helm.internal.createindex.CreateIndexFacet;
+import org.sonatype.repository.helm.internal.hosted.HelmHostedFacet;
+import org.sonatype.repository.helm.internal.metadata.ChartIndex;
+import org.sonatype.repository.helm.internal.orient.createindex.HelmIndexInvalidationEvent;
+import org.sonatype.repository.helm.internal.util.YamlParser;
+
+import com.google.common.eventbus.AllowConcurrentEvents;
+import com.google.common.eventbus.Subscribe;
+import org.joda.time.DateTime;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.common.stateguard.StateGuardLifecycleSupport.State.STARTED;
+import static org.sonatype.repository.helm.internal.AssetKind.HELM_INDEX;
+
+/**
+ * Facet for rebuilding Helm index.yaml files
+ *
+ * @since 1.0.11
+ */
+@Named
+public class CreateIndexFacetImpl
+    extends FacetSupport
+    implements CreateIndexFacet, Asynchronous
+{
+  private final EventManager eventManager;
+
+  private final long interval;
+
+  private static final String INDEX_YAML = "index.yaml";
+
+  private static final String INDEX_YAML_CONTENT_TYPE = "text/x-yaml";
+
+  private final AtomicBoolean acceptingEvents = new AtomicBoolean(true);
+
+  private final static String API_VERSION = "v1";
```

Comment:
Modifiers should be declared in the correct order.

Reply:
fixed 6aa74c061ac979c8caed17119faa1b450e8e0327